### ISSUE TITLE
Displaying character names under images

### ISSUE THREAD

**Issue by memorydata** [antialiasis/favorite-picker#12]
Hello! I'm very new to coding or writing HTML and I don't do it often, but I wanted to make a favorites picker.

The picker that I'm making includes characters that don't have a "canon" design, since they're from a book series, and thus, any images I use won't be instantly recognizable.

I was curious if there's a way to make it so that the name of each character could be displayed underneath each character, alongside the circle portraits.

Also, I noticed that on the Favorite Pokemon Picker, the Pokemon are overlaid over the circle, rather than the edges of the image being clipped by the circle. Is there a setting/anything I can add to make it so my picker does the same thing?

Thank you so much, I don't know if this project is still actively maintained but nonetheless this has been a big help for starting my own picker!

**Comment by antialiasis** [comment]
Hey! That's absolutely possible - you just need to override the `getItemElem` function. Something like this:

```
var pickerUI = new PickerUI(myPicker, {
    elements: {
        pick: "#pick",
        pass: "#pass",
        undo: "#undo",
        redo: "#redo",
        evaluating: "#evaluating",
        favorites: "#favorites"
    },
    getItemElem: function(item, settings) {
        return $('<li><span class="image"><span><img src="' + item.image + '" alt="' + item.name + '" title="' + item.name + '" loading="eager"></span></span><span class="name">' + item.name + '</span></li>');
    }
});
```

Meanwhile, this CSS should make names and images look nice together and make the border overlap:

```
.item-list li {
    margin:2px;
    display:inline-block;
    padding:0;
    text-align:center;
    vertical-align:top;
    line-height:1.2;
    width:106px;
}

.item-list li .name {
    line-height:1.2;
}

.item-list li .image {
    display:block;
    height:106px;
    width:106px;
    background:#F7F7F7;
    border-radius:100%;
    z-index:1;
    position:relative;
    overflow:hidden;
}

.item-list li img {
    position:absolute;
    top:-3px;
    left:-3px;
    z-index:2;
}

.item-list li .image span {
    padding:3px;
}

.item-list li.selected .image span {
    display:block;
    height:100px;
    width:100px;
    padding:0;
    border:3px solid #7AC;
    border-radius:100%;
}

.item-list li:hover .image {
    background:#CDF;
}

.item-list.sorting li:hover .image {
    background:#F7F7F7;
}
```

You'll want to remove the corresponding CSS that's already there, including this chunk that's a bit below the rest in the sample file:

```
.item-list li:hover {
    background:#CDF;
}

.item-list.sorting li:hover {
    background:#F7F7F7;
}

#evaluating li.selected {
    border:3px solid #7AC;
}
```

Let me know if you have further trouble with it.

**Comment by memorydata** [comment]
Hi, thank you so much, especially with how quickly you responded! I really appreciate it. 
![image](https://user-images.githubusercontent.com/87771895/126562391-bc68c77a-35e7-4a0b-8eb3-2166925d1ec1.png)
This is currently how it ended up looking. My main problem now is resizing the images, since I know the previous code would automatically resize them to fit in the confines of the image. More than likely, I probably made an error when replacing the previous CSS and/or the new CSS doesn't have the resizing, which would make sense since I assume the overlap can't coincide with the image resizing. (Apologies that most of them don't have images yet, I've only added a few to test it.)

I ended up deciding to crop the artwork I was using into headshots, which I believe should look better without the border overlap, but I also realize how indecisive I sound most likely LOL.

[Here is the website I'm using to test the code,](https://wcpicker.neocities.org/picker.html) and [here's a Pastebin to the code itself, since as I said before, there's a pretty big chance that I messed up along the way.](https://pastebin.com/kczEhn3u)

Once again, thank you so much for your help!
